.github/workflows/ci-preview.yml

```diff
@@ -0,0 +1,29 @@
+name: Generate Preview (CI)
+
+concurrency:
+  group: ci-${{ github.event.number }}
+  cancel-in-progress: true
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build-preview:
+    if: ${{ github.repository_owner == 'cryostatio' }}
+    env:
+      DEPLOY_DOMAIN: ${{ github.repository_owner }}-main.surge.sh
```

Comment:
Perhaps this should be `cryostatio-cryostat-web-main` or something like that, ie `owner-repo-branch`. I can imagine more cases where we deploy things to Surge, like API docs for various other components.